File: system-test/typescript/src/testkit/driver/port/external/tax/tax-driver.ts
import { Result } from '../../../../common/result.js';
import {
  TaxErrorResponse,
  GetTaxResponse,
  ReturnsTaxRateRequest,
} from '../../../../common/dtos.js';

export interface TaxDriver {
  goToTax(): Promise<Result<void, TaxErrorResponse>>;
  getTaxRate(country: string): Promise<Result<GetTaxResponse, TaxErrorResponse>>;
  returnsTaxRate(request: ReturnsTaxRateRequest): Promise<Result<void, TaxErrorResponse>>;
  close(): Promise<void>;
}

File: system-test/java/src/main/java/com/optivem/shop/testkit/driver/port/external/tax/TaxDriver.java
package com.optivem.shop.testkit.driver.port.external.tax;

import com.optivem.shop.testkit.driver.port.external.tax.dtos.GetTaxResponse;
import com.optivem.shop.testkit.driver.port.external.tax.dtos.ReturnsTaxRateRequest;
import com.optivem.shop.testkit.driver.port.external.tax.dtos.error.TaxErrorResponse;
import com.optivem.shop.testkit.common.Result;

public interface TaxDriver extends AutoCloseable {
    Result<Void, TaxErrorResponse> goToTax();

    Result<GetTaxResponse, TaxErrorResponse> getTaxRate(String country);

    Result<Void, TaxErrorResponse> returnsTaxRate(ReturnsTaxRateRequest request);
}
